### PR TITLE
[ng] single-selection for datagrid

### LINKED ISSUE #475: Implement single-selection for datagrid
**Select one ...**  (check one with "x")
```
[ ] bug
[x] feature request
[ ] enhancement
```

### Expected behavior

We need to allow datagrids that are selectable but only allow one selection at a time.

### Actual behavior

Currently the only options are 1. not selectable, or 2. multi-selection. There's no way to enforce single-selection.

### Environment details

* **Angular version:** 2.0.X

* **Clarity version:** All

* **OS and version:** All

* **Browser:** All

### PR DESCRIPTION
**Pull request by jeeyun**
Single-select only mode in a datagrid.

screen shot 2017-02-17 at 1 20 09 pm

resolves #475.

Signed-off-by: Jeeyun Lim <jeeyun.lim@gmail.com>

### CODE REVIEW

File: src/clarity-angular/datagrid/selection-type.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c) 2016 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+export enum SelectionType {
```
Comment (youdz) [inline src/clarity-angular/datagrid/selection-type.ts]: This feels out of place here. I think you should just put this enum in `providers/selection.ts`, that's the most logical place for it.

I know it's good practice is to have one class per file, but constants, enums or small interfaces directly linked to a class can be in the same file without it becoming pollution. It actually makes things clearer.

File: src/clarity-angular/datagrid/datagrid.ts [hunk 2]
```diff
@@ -177,7 +193,14 @@ export class Datagrid implements AfterContentInit, AfterViewInit, OnDestroy {
...
         this._subscriptions.push(this.page.change.subscribe(() => this.triggerRefresh()));
-        this._subscriptions.push(this.selection.change.subscribe(s => this.selectedChanged.emit(s)));
+        this._subscriptions.push(this.selection.change.subscribe(s => {
+            if (this.selection.selectionType === SelectionType.Single) {
+                this.singleSelectedChanged.emit(s);
+            } else if (this.selection.selectionType === SelectionType.Multi) {
+                this.selectedChanged.emit(s);
+            }
+        }));
+        // this._subscriptions.push(this.selection.change.subscribe(s => this.singleSelectedChanged.emit(s)));
```
Comment (jeeyun) [inline src/clarity-angular/datagrid/datagrid.ts]: Oops. No, good catch. Removing.

File: src/clarity-angular/datagrid/index.ts [hunk 3]
```diff
@@ -35,6 +35,8 @@ export * from "./interfaces/filter";
 export * from "./interfaces/string-filter";
 export * from "./interfaces/comparator";
 
+export * from "./selection-type";
```
Comment (jeeyun) [inline src/clarity-angular/datagrid/index.ts]: Fixed.

File: src/app/datagrid/selection-single/selection-single.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
+<!--
+  ~ Copyright (c) 2016 VMware, Inc. All Rights Reserved.
+  ~ This software is released under MIT license.
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+
+<h2>Selection (Single)</h2>
+
+<p>
+    To only allow selecting one row at a tune, we provide the ability for the user to
```
Comment (jeeyun) [inline src/app/datagrid/selection-single/selection-single.html]: Haha. Thank you for catching that.

File: src/app/datagrid/selection-single/selection-single.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    do so. To make rows selectable in your datagrid, you need to do the following:
```
Comment (jeeyun) [inline src/app/datagrid/selection-single/selection-single.html]: Done.

File: src/clarity-angular/datagrid/selection-type.ts (hunk 1 above)
Comment (jeeyun) [inline src/clarity-angular/datagrid/selection-type.ts]: Done.

**Pull request merged**